src/peer_conn_protocol.rs

```diff
@@ -5,32 +5,75 @@ use super::server;
 use std::io;
 use std::sync::mpsc::Sender;
 use uuid::Uuid;
+use std::collections::HashMap;
+use std::net::TcpStream;
 
-pub fn send_p2p_handshake(
-    conn_id: u64,
-    peer_address: String,
-    peer_port: u16,
-    messenger: Sender<MessengerOperations>,
-) -> Result<Uuid, io::Error> {
-    let stream = server::new_connection(peer_address.clone(), peer_port)?;
-    messenger
-        .send(MessengerOperations::New(NewConnectionMessage {
+pub struct Peer {
+    pub peer_address: String,
+    pub peer_port: u16,
+    pub stream: TcpStream,
+}
+
+impl Peer {
+    pub fn send_p2p_handshake(
+        &mut self,
+        conn_id: u64,
+        messenger: Sender<MessengerOperations>,
+    ) -> Result<Uuid, io::Error> {
+        
+        messenger
+            .send(MessengerOperations::New(NewConnectionMessage {
+                conn_id,
+                socket: self.stream.try_clone().unwrap(),
+            }))
+            .unwrap();
+    
+        send_packet!(
+            messenger,
             conn_id,
-            socket: stream.try_clone().unwrap(),
-        }))
+            Packet::Handshake(Handshake {
+                protocol_version: 404,
+                server_address: self.peer_address.clone(),
+                server_port: self.peer_port,
+                next_state: 6,
+            })
+        )
         .unwrap();
+    
+        Ok(Uuid::new_v4())
+    }
+}
 
-    send_packet!(
-        messenger,
-        conn_id,
-        Packet::Handshake(Handshake {
-            protocol_version: 404,
-            server_address: peer_address.clone(),
-            server_port: peer_port,
-            next_state: 6,
-        })
-    )
-    .unwrap();
-
-    Ok(Uuid::new_v4())
+pub struct PeerManager {
+    pub peer_map: HashMap::<u64, Peer>,
 }
+
+impl PeerManager{
+    pub fn new() -> PeerManager{
+        let peer_map = HashMap::<u64, Peer>::new();
```

Comment:
I dont think this belongs in this file--it's unrelated to the protocol

Reply:
What will be referring to this state in the future? Might be helpful for me to understand what this state is for

Reply:
I was thinking about renaming it peer_management.rs I think it would make more sense. The state is there to keep track of everything, we could just iterate over this map to send messages later on or we could use it to delete connections between peers.

Reply:
What messages would we send over this? Who would be sending them?

Reply:
The server itself could send packets through these tcpstream later. Game state packets for example.

Reply:
Why would we send a gamestate packet just to peers? What would be sending a game state packet that would need to know about the existence of peers?

Reply:
I think if we don't know what this is for, or if its use will be way in the future, we shouldn't add it. Additionally I think that we should avoid using shared memory wherever possible and instead use messenging APIs. So if we were to implement something like this it should follow the pattern of the messenger. I don't see any reason why that would be a problem with this module since it doesn't return any meaningful results